src/integration-test/java/io/aiven/kafka/connect/http/ErrantRecordTest.java

```diff
@@ -0,0 +1,341 @@
+            "org.apache.kafka.common.serialization.ByteArrayDeserializer",
+            ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+            "org.apache.kafka.common.serialization.ByteArrayDeserializer",
+            ConsumerConfig.MAX_POLL_RECORDS_CONFIG, "1",
+            ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"
+        );
+
+        consumer = new KafkaConsumer<>(consumerProps);
+        consumer.subscribe(List.of(TEST_DLQ_TOPIC));
+
+        final List<String> expectedBodies = new ArrayList<>();
+        final List<String> errantBodies = new ArrayList<>();
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        for (int i = 0; i < 15; i++) {
+            for (int partition = 0; partition < TEST_TOPIC_PARTITIONS; partition++) {
+                final String key = "key-" + i;
+                final var recordName = "user-" + i;
+                final var recordValue = "value-" + i;
+                final Record value = new Record(recordName, recordValue);
+                if (i % FAIL_PERIOD == 0) {
+                    errantBodies.add(String.format(JSON_PATTERN, recordName, recordValue));
+                } else {
+                    expectedBodies.add(String.format(JSON_PATTERN, recordName, recordValue));
+                }
+                sendFutures.add(sendMessageAsync(partition, key, value));
+            }
+        }
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(10)).pollInterval(Duration.ofMillis(100))
+                .until(() -> failingPeriodicHandler.recorderBodies().size() >= expectedBodies.size());
+
+        assertThat(failingPeriodicHandler.recorderBodies()).containsExactlyElementsOf(expectedBodies);
+
+        log.info("{} HTTP requests were expected, {} were successfully delivered",
+            expectedBodies.size(),
+            failingPeriodicHandler.recorderBodies().size());
+
+        // read from the DLQ
+        final int readConsumerRetries = 100;   
+        int noRecordsCount = 0;
+        final List<String> dlqBodies = new ArrayList<>();
+
+        while (noRecordsCount < readConsumerRetries) {
+            final ConsumerRecords<String, Record> consumerRecords = 
+                    consumer.poll(Duration.ofMillis(100));
+
+            if (consumerRecords.count() == 0) {
+                noRecordsCount++;
+                continue;
+            }
+
+            for (final ConsumerRecord<String, Record> record : consumerRecords) {
+                dlqBodies.add(String.format(JSON_PATTERN, record.key(), record.value()));
+            }
+            consumer.commitAsync();
+        }
+        consumer.close();
+
+        // checking that all failed messages were on the DLQ
+        assertThat(errantBodies.containsAll(dlqBodies));
```

Comment:
Perhaps you wanted to write something like
```suggestion
        assertThat(errantBodies).containsExactlyInAnyOrderElementsOf(dlqBodies);
```
because right now the assertion does not actually check anything.
Also, after fixing the assertion the test fails. Could you please fix it?

Reply:
Fixed in latest commit.